commands/http/handler.go

```diff
@@ -55,6 +56,20 @@ func (i Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	log.Debug("Incoming API request: ", r.URL)
 
+	// error on external referers (to prevent CSRF attacks)
+	referer := r.Referer()
+	scheme := r.URL.Scheme
+	if len(scheme) == 0 {
+		scheme = "http"
+	}
+	host := fmt.Sprintf("%s://%s/", scheme, r.Host)
+	// empty string means the user isn't following a link (they are directly typing in the url)
+	if referer != "" && !strings.HasPrefix(referer, host) {
```

Comment:
Hmm, wont this break sharness, too? or curling? oh i guess those dont set the referrer? The security of the API will probably have to move towards tokens or something in the future.